lib/graphql/query/context.rb

```diff
@@ -163,15 +164,30 @@ def initialize(query:, values: , object:)
       # @api private
       attr_writer :value
 
-      def_delegators :@provided_values, :[], :[]=, :to_h, :to_hash, :key?, :fetch, :dig
-      def_delegators :@query, :trace, :interpreter?
+      # @api private
+      attr_accessor :scoped_context
 
-      # @!method [](key)
-      #   Lookup `key` from the hash passed to {Schema#execute} as `context:`
+      def_delegators :@provided_values, :[]=
```

Comment:
🤔 is it inconsistent that the getter `[]` will magically return scoped values but the setter `[]=` won't?

Reply:
The API could be a little confusing if the developer isn't aware of the existence of `#scoped_merge!`. Using a combination of `#[]=` and `#scoped_merge` can also lead to misleading behaviour with `#[]` because it always retrieves the value from `@scoped_context` if it exists regardless of the existing value in `@provided_values`, as described in https://github.com/rmosolgo/graphql-ruby/pull/2634#issuecomment-563289122

```ruby
context.scoped_merge!(value: 'old')
context[:value] = 'new'
assert_equal('old', context[:value])
```

I can definitely see how this usage of the API can be misleading. I could also see this more as an edge case though. This only arises when values are set in ☝️ configuration, and using `scoped_merge!(special_key: 1)` in combination with `[:special_key] = 2` is probably not what that developer _means_ to do, because by doing `[:special_key] = 2`, they are voiding the benefit of having `:special_key` be scoped.

Reply:
Yeah I think what you have makes sense 👍 

Also wondering if `set_scoped(key, value)` should exist in addition to `scoped_merge!` as a convenience. If a common use case is setting a single key then it might match people's normal usage of setting a hash key?

Meaning most people would use `[]=` instead of `merge`

Reply:
Yep! I like `set_scoped(key, value)` as a convenience. I'll push a commit shortly.

Reply:
Added `Context#scoped_set(key, value)` in commit https://github.com/rmosolgo/graphql-ruby/pull/2634/commits/afc292af7ca85762eea59b7693b1c0c74dad2a6b.